rest-service/manager_rest/resources.py

```diff
@@ -1176,27 +1176,42 @@ def get(self, deployment_id, **_):
                                            outputs=outputs)
 
 
+def _elasticsearch_connection():
+    es_host = config.instance().db_address
+    es_port = config.instance().db_port
+    return elasticsearch.Elasticsearch(hosts=[{"host": es_host,
+                                              "port": es_port}])
+
+
+def _check_index_exists(index_name):
+    es = _elasticsearch_connection()
+    return es.indices.exists(index=[index_name])
+
+
 def _query_elastic_search(index=None, doc_type=None, body=None):
     """Query ElasticSearch with the provided index and query body.
 
     Returns:
     Elasticsearch result as is (Python dict).
     """
-    es_host = config.instance().db_address
-    es_port = config.instance().db_port
-    es = elasticsearch.Elasticsearch(hosts=[{"host": es_host,
-                                             "port": es_port}])
+    es = _elasticsearch_connection()
     return es.search(index=index, doc_type=doc_type, body=body)
 
 
 class Events(SecuredResource):
 
+    def _set_index_name(self):
+        if _check_index_exists('cloudify_events'):
+            return 'cloudify_events'
+        else:
+            return 'logstash-*'
+
     def _query_events(self):
         """
         List events for the provided Elasticsearch query
         """
         verify_json_content_type()
-        return _query_elastic_search(index='cloudify_events',
+        return _query_elastic_search(index=self._set_index_name(),
```

Comment:
research flask.g object to store both es connection object and make the index exists check only once.